lib/ddtrace/propagation/grpc_propagator.rb

```diff
@@ -0,0 +1,55 @@
+require 'ddtrace/context'
+require 'ddtrace/ext/distributed'
+
+module Datadog
+  # opentracing.io compliant methods for distributing trace context
+  # between two or more distributed services. Note this is very close
+  # to the HTTPPropagator; the key difference is the way gRPC handles
+  # header information (called "metadata") as it operates over HTTP2
+  module GRPCPropagator
+    include Ext::DistributedTracing
+
+    def self.inject!(context, metadata)
+      metadata[HTTP_HEADER_TRACE_ID] = context.trace_id.to_s
```

Comment:
I appreciate the idea of being consistent with the other context propagation headers, but would it make sense to create a new set of constants for gRPC? e.g. `GRPC_METADATA_TRACE_ID`/`GRPC_TRACE_ID`? Even if the values are roughly the same or equivalent?

Might be a good idea to avoid coupling with HTTP implementation in case there's some divergence in implementation later.